src/lua/oltp_common.lua

```diff
@@ -270,6 +290,60 @@ CREATE TABLE sbtest%d(
    end
 end
 
+function bulk_inserts(con, table_num)
+
+   if (sysbench.opt.table_size > 0) then
+      time = os.date("*t")
+   end
+   iquery = ""
+
+   if sysbench.opt.auto_inc then
+      iquery = "INSERT INTO sbtest" .. table_num .. "(k, c, pad) VALUES"
+   else
+      iquery = "INSERT INTO sbtest" .. table_num .. "(id, k, c, pad) VALUES"
+   end
+
+   con:bulk_insert_init(iquery)
+
+   local c_val
+   local pad_val
+
+   cursize = 1
+
+   for i = 1, sysbench.opt.batch_insert_count do
+
+       if (sysbench.opt.batch_insert_count <=2000) then
+           c_val = get_c_value()
+           pad_val = get_pad_value()
+       else
+           c_val = get_c_value_smaller()
+           pad_val = get_pad_value_smaller()
+       end
+
+       if (sysbench.opt.auto_inc) then
+          query = string.format("(%d, '%s', '%s')",
+                                sysbench.rand.default(1, sysbench.opt.table_size),
+                                c_val, pad_val)
+       else
+          i = sysbench.rand.unique() - 2147483648
```

Comment:
will not work as you are modifying i

Reply:
Scope of "i" dont impact the loop. Though I have changed the logic here. 

The logic of `sysbench.rand.unique() - 2147483648` does not work in this case since it results in conflict for few key which are part of 4k batch(or lesser) as in load phase we have already inserted 50M rows. 

With new logic, I increment the row count serially above the #rows inserted part of load phase.